### PR TITLE
Add commands extensive docu

### PR DESCRIPTION
**Pull request by neithernut**
This PR adds more extensive documentation to the command subsystem.

A view things may still be missing from the documentation. Please review and annotate uncertainties you may find.

### CODE REVIEW

File: src/command/command.h [hunk 1]
```diff
@@ -57,7 +57,40 @@ typedef int (*ws_special_command_func)(struct ws_processor*,
 /**
  * Command type
  *
- * This datatype represents a command, which may be invoked in a transaction.
+ * This type represents a command exposed via the API.
+ * Commands may be invoked in a transactions.
+ *
```
Comment (xVin) [inline src/command/command.h:62]: either remove the "a" or change it to "transaction"

File: src/command/command.h [hunk 2]
```diff
@@ -57,7 +57,40 @@ typedef int (*ws_special_command_func)(struct ws_processor*,
...
+ * don't have any side effects concerning the processor's operation or state.
+ * Arguments and results are passed via a stack @see stack_semantics.
+ * Preparation of the stack is done by the `ws_action_processor`.
+ *
+ * ### Special commands
+ *
+ * Special commands _do_ have side effects on the processor.
+ * E.g. a jump will increment a processor's (virtual) program counter.
+ *
+ * Another property of a special command it that is _should not_ have any effect
```
Comment (xVin) [inline src/command/command.h]: it --> is
is --> it

File: src/command/command.h [hunk 3]
```diff
@@ -57,7 +57,40 @@ typedef int (*ws_special_command_func)(struct ws_processor*,
...
+ * on the stack, _unless_ the command is intended to alter the stack itself.
+ * E.g. a `jump` will have no effect on the stack while a `push` will _only_
+ * affect the stack.
+ * Because of this, parameter passing is done in a completely different way.
+ * Instead of a prepared stack, the command will be passed a raw argument list,
+ * along with the processor's context.
```
Comment (xVin) [inline src/command/command.h]: did you mean "a raw argument list will be passed to the command"?

File: src/command/statement.h [hunk 4]
```diff
@@ -40,8 +40,28 @@ struct ws_command;
 /**
  * A command argument
  *
- * There are two types of arguments: the ones passed directly and the ones which
- * are already on the stack.
+ * Naturally arguments may be passed to commands.
+ * This type represents an arguments being passed to a command which may be
```
Comment (xVin) [inline src/command/statement.h]: argument

File: src/command/command.h [hunk 5]
```diff
@@ -57,7 +57,40 @@ typedef int (*ws_special_command_func)(struct ws_processor*,
...
+ *
+ * There are two types of commands:
+ *  * Regular commands
+ *  * Special commands
+ *
+ * ### Regular commands
+ *
+ * Regular commands are commands which consume arguments, have one result and
+ * don't have any side effects concerning the processor's operation or state.
+ * Arguments and results are passed via a stack @see stack_semantics.
```
Comment (matthiasbeyer) [inline src/command/command.h]: Have you compiled the documentation and reviewed this `@see`? I don't know in what it results, but maybe it results in 

```
via a stack stack_semantics
```

or something.

File: src/command/command.h (hunk 3 above)
Comment (neithernut) [inline src/command/command.h]: Mh... I **really** fucked up that sentence.

File: src/command/command.h (hunk 5 above)
Comment (neithernut) [inline src/command/command.h]: Nope, it generated a proper "See also" section, which was in the completely wrong place.
Turns out doxygen isn't smart enough to perform inline links _or_ at least group all the refs together on it's own.

The link isn't created properly, either.

**Comment by neithernut** [comment]
Note: Some linking issue is still unresolved.

**Comment by neithernut** [comment]
The linking issue is resolved.

**Comment by matthiasbeyer** [comment]
What linking issue? You mean the `size_t` vs `ssize_t` ?

**Comment by neithernut** [comment]
Linking issues in the _documentation_. (Refs)

**Comment by matthiasbeyer** [comment]
Ah, of course.

ACK.

**Pull request merged**